Sources/CasePaths/EnumReflection.swift

```diff
@@ -67,7 +67,10 @@ public func extract<Root, Value>(case embed: (Value) -> Root, from root: Root) -
       any = anyChild
     }
     if MemoryLayout<Value>.size == 0, !isUninhabitedEnum(Value.self) {
-      return (["\(root)"], unsafeBitCast((), to: Value.self))
```

Comment:
This code path should only hit for enum cases with no associated data, e.g.:

```swift
enum Foo {
  case bar
}
```

It sounds like the stringified enum cases were more complex and had more data, right?

Is there a chance you can recreate a poorly-performing extraction? We could get a benchmark in place using [swift-benchmark](https://github.com/google/swift-benchmark) to get a better handle on how to improve performance.

Reply:
Oh, I see, this code path also hits for an embedded class/struct in the deepest code path. Lemme get a benchmark in place to see if we can get a check for this.

Reply:
Actually ha, this is still for things with zero memory, which wouldn't be most structs. We'd still really appreciate an example data type we can benchmark 😄

Reply:
Hello Stephen,

It hits when you're trying to match some case without associated value with another case with associated value from a sibling enum. For example, the following test passes, but hits L70 with `root` being 42:
```
func testSiblingCases() {
  enum Parent {
    case foo(Foo)
    case bar(Bar)
  }
  enum Foo {
    case value(Int)
  }
  enum Bar {
    case unrelated
  }
  
  XCTAssertNil(
    extract(case: { Parent.foo(.value(42)) }, from: Parent.bar(.unrelated))
  )
}
```

Reply:
I've just roughly tested the Hashable/nonHashable cases using a 100k chars string in `Foo.value` and the `Hashable` route is ~1000x faster. Proper benchmarks should show interesting results.

Please note that in order to enter the `Hashable` branch, `Parent` has to be `Hashable`. Automatic conformance induces that all child enums and their associated values should be `Hashable` too.